Add comprehensive system tray menu and submenu icon examples to documentation

s00d opened the pull request:
#### Description

I'm currently developing a cross-platform system tray application [SwitchShuttle](https://github.com/s00d/switchshuttle) and have been extensively working with Tauri's system tray menu API. During development, I noticed that the current documentation could benefit from more practical examples showing real-world usage patterns.

This PR adds several comprehensive examples to the system tray documentation, including:

- **Dynamic menu updates** - How to update menu item text in real-time
- **Background monitoring** - Examples of updating menu items from background threads
- **JavaScript integration** - How to handle menu updates from the frontend
- **Real-time data display** - Practical examples of showing live system information in menu items

s00d committed:
docs: Update system-tray.mdx with comprehensive Tauri tray API documentation

Legend-Master commented:
I know the page order in the side bar made it hard to spot, but I think most of the content are talking about menus and should probably be put in https://tauri.app/learn/window-menu/ instead, and we can add a link in https://tauri.app/learn/system-tray/ pointing to it

s00d committed:
docs(system-tray): remove outdated menu items section and link to docs

s00d committed:
docs(window-menu): enhance documentation on window menus and item types

s00d committed:
feat(docs): add submenu icon support in window-menu documentation

s00d committed:
docs(window-menu): remove unnecessary whitespace in documentation

s00d commented:
> I know the page order in the side bar made it hard to spot, but I think most of the content are talking about menus and should probably be put in https://tauri.app/learn/window-menu/ instead, and we can add a link in https://tauri.app/learn/system-tray/ pointing to it

@Legend-Master Hi, I moved the documentation to the `system-tray` section and added descriptions for submenu icons.
These changes can be included in the 2.8.0 release.

vasfvitor commented:
lmk when it's ok to merge, I'll then go and fix the building (prolly needs to rebase/update branch), checks (pnpm format) and merge

s00d committed:
docs(window-menu): add note on icon support for submenus in Tauri 2.8.0

s00d committed:
docs(system-tray): remove redundant information about menus

s00d committed:
docs(system-tray): add link to window menu documentation

s00d committed:
feat(window-menu): update submenu icons with dynamic images

s00d commented:
> lmk when it's ok to merge, I'll then go and fix the building (prolly needs to rebase/update branch), checks (pnpm format) and merge

Added a couple more fixes, looks like I’m done.
Strange that the build failed.

vasfvitor committed:
Merge branch 'v2' into patch-1

vasfvitor committed:
format

vasfvitor commented:
@Legend-Master awaiting your approval, thanks

Legend-Master commented on src/content/docs/learn/window-menu.mdx:
```diff
@@ -7,7 +7,7 @@ i18nReady: true
 import { Tabs, TabItem } from '@astrojs/starlight/components';
 
-Native application menus can be attached to both to a window or system tray. Available on desktop.
+Native application menus can be attached to windows. This documentation covers menu creation, management, and event handling for window menus. Available on desktop platforms.
```
To be honest, I personally like the old description better

Legend-Master commented on src/content/docs/learn/window-menu.mdx:
```diff
@@ -266,6 +281,12 @@ fn main() {
             app.set_menu(menu)?;
 
+            let menu_image_update = Image::from_bytes(include_bytes!("../icons/menu_update.png")).unwrap();
+            // You can also update the submenu icon dynamically
+            file_menu.set_icon(Some(menu_image_update))?;
+            // Or set a native icon
+            file_menu.set_native_icon(Some(tauri::menu::NativeIcon::Folder))?;
```
We should note that only one of them applies? (could use a platform config to use native icon on macOS and `set_icon` on others)

Legend-Master commented on src/content/docs/learn/window-menu.mdx:
```diff
@@ -548,5 +569,372 @@ fn main() {
+</TabItem>
+</Tabs>
+
+### Updating Menu Item Text
```
I think the examples inside this section are currently the same as the one above?

Legend-Master commented on src/content/docs/learn/window-menu.mdx:
```diff
@@ -548,5 +569,372 @@ fn main() {
+</TabItem>
+</Tabs>
+
+### Submenus with Icons
```
I think we have this part covered in `#creating-a-multi-level-menu` is enough, since the syntax is basically the same as a normal menu item, we don't need to cover it in a stand alone section

Legend-Master reviewed (commented):
To be honest, I feel like most of the new docs could be merged inside the existing docs, introducing new sections made it quite repetitive

s00d committed:
docs(window-menu): update documentation for menu attachment and icons

s00d committed:
docs(window-menu): remove unnecessary blank lines for clarity

s00d committed:
docs(window-menu): update menu item descriptions and examples

s00d commented:
> To be honest, I feel like most of the new docs could be merged inside the existing docs, introducing new sections made it quite repetitive

@Legend-Master Hello, I removed all the new sections — added the new info as code examples in the existing ones.

Legend-Master committed:
Fix examples

Legend-Master committed:
Remove extra clone

Legend-Master reviewed (approved):
Thanks
